### PR TITLE
Clear edit queue when all evaluations are interrupted

### LINKED ISSUE #978: Stop printing to the output window when all evaluations are interrupted
# How it happened

Accidentally evaluated a function that ended up outputting about 5k lines worth (And each line was of substantial length as well) of material onto the REPL. Interrupting evaluation didn't work because it was already evaluated, so it's a bottleneck of printing.

# Thoughts
The following is a summary of what @bpringe has said to me about the matter:

The output window queues up prints and uses the `applyEdit` function of the VSCode API. Because `applyEdit` is async, it happens after evaluation, and it's slow in general, which is why printing seems to happen in chunks with what appears to be a pause in between each chunk.

The solution will likely lie in clearing this queue, which should be pretty easy because it's just an array.

### PR DESCRIPTION
**Pull request by bpringe**
## What has Changed?

- Clear edit queue for output window when the "Interrupt all evaluations" command is run, so that output does not continue to print after evaluations are interrupted

Closes #978 

## My Calva PR Checklist

I have:

- [x] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [x] Directed this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [x] Made sure I have changed the PR base branch, so that it is not `published`. (Sorry for the nagging.)
- [x] Updated the `[Unreleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.
- [x] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and either tested it there if so, or mentioned it in the PR.
- [x] Tested the VSIX built from the PR (so, after you've submitted the PR). You'll find the artifacts by clicking _Show all checks_ in the CI section of the PR page, and then _Details_ on the `ci/circleci: build` test. NB: *There is a CircleCI bug that makes the Artifacts hard to find. Please see [this issue](https://discuss.circleci.com/t/artifacts-tab-not-showing-unless-logged-in/32433) for workarounds.*
     - [x] Tested the particular change
     - [x] Figured if the change might have some side effects and tested those as well.
     - [x] Smoke tested the extension as such.
- [x] Referenced the issue I am fixing/addressing _in a commit message for the pull request_.
     - [x] If I am fixing the issue, I have used [GitHub's fixes/closes syntax](https://help.github.com/en/articles/closing-issues-using-keywords)
     - [x] If I am fixing just part of the issue, I have just referenced it w/o any of the "fixes” keywords.
- [x] Created the issue I am fixing/addressing, if it was not present.
- [x] Added to or updated docs in this branch, if appropriate

## The Calva Team PR Checklist:

Before merging we (at least one of us) have:

- [ ] Made sure the PR is directed at the `dev` branch (unless reasons).
- [ ] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and tested it there if so.
- [ ] Read the source changes.
- [ ] Given feedback and guidance on source changes, if needed. (Please consider noting extra nice stuff as well.)
- [ ] Tested the VSIX built from the PR (well, if this is a PR that changes the source code.)
     - [ ] Tested the particular change
     - [ ] Figured if the change might have some side effects and tested those as well.
     - [ ] Smoke tested the extension as such.
- [ ] If need be, had a chat within the team about particular changes.

Ping @pez, @kstehn, @cfehse, @bpringe

### CODE REVIEW

File: src/evaluate.ts [hunk 1]
```diff
@@ -34,6 +33,10 @@ function interruptAllEvaluations() {
         } else {
             vscode.window.showInformationMessage('Interruption command finished (unknown results)');
         }
+        if (outputWindow.getEditQueueLength() > 0) {
+            outputWindow.clearEditQueue();
+            outputWindow.appendPrompt();
```
Comment (PEZ) [inline src/evaluate.ts]: Maybe the edit queue shouldn't be exported from the outputWindow module? Seems like it is an internal concern.

File: src/results-output/results-doc.ts [hunk 2]
```diff
@@ -303,6 +303,14 @@ export function append(text: string, onAppended?: OnAppendedCallback): void {
...
 }
 
+export function clearEditQueue(): void {
+    editQueue = [];
+}
+
+export function getEditQueueLength(): number {
+    return editQueue.length;
+}
+
```
Comment (PEZ) [inline src/results-output/results-doc.ts]: An exported function named something like `stopPrinting()` makes more sense to me. Or maybe that makes you think there should be a `resumePrinting()` as well... Hmmm maybe `discardPendingPrints()` and that function would take care of the business.

**Review by PEZ — APPROVED** [review]

**Comment by PEZ** [comment]
I had some minor opinions. 😄

File: src/evaluate.ts (hunk 1 above)
Comment (bpringe) [reply src/evaluate.ts]: This is why I didn't export the actual queue, only a couple of functions around it. I'd say in some cases, like this, it is a concern of calling code, as the interrupt function needs to know if clearing the queue and appending a prompt is necessary. I'm open to suggestions though if you feel this code should be refactored.

Edit: See my reply to your comment on those functions. I saw that after I said the above.

File: src/results-output/results-doc.ts (hunk 2 above)
Comment (bpringe) [reply src/results-output/results-doc.ts]: Oh I see what you mean now. I'll make changes.

**Pull request merged**